src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionPerfTests.cs

```diff
@@ -656,5 +656,60 @@ class C
             var comp = CreateCompilation(sourceBuilder.ToString());
             comp.VerifyDiagnostics();
         }
+
+        [ConditionalFact(typeof(IsRelease))]
+        public void DefiniteAssignment_ManySwitchCasesAndLabels()
+        {
+            const int nLabels = 1500;
+
+            var builder = new StringBuilder();
+            builder.AppendLine("#nullable enable");
```

Comment:
Consider extracting a simple version of this generated source (e.g. with nLabels = 5 or something) as a standalone test to make it easier to conceptualize the general scenario we are addressing.

Reply:
Added a comment with the source.